components/credentials/src/main/kotlin/org/cloudfoundry/credhub/domain/CertificateGenerationParameters.kt

```diff
@@ -90,15 +90,19 @@ class CertificateGenerationParameters : GenerationParameters {
         }
 
         val that = other as CertificateGenerationParameters?
-        return keyLength == that!!.keyLength &&
-            duration == that.duration &&
-            isSelfSigned == that.isSelfSigned &&
-            isCa == that.isCa &&
-            (caName == that.caName || caName == null || that.caName == null) &&
-            X500Name(that.x500Principal!!.name) == X500Name(this.x500Principal!!.name) &&
-            alternativeNames == that.alternativeNames &&
-            extendedKeyUsage == that.extendedKeyUsage &&
-            keyUsage == that.keyUsage
+        return duration == that!!.duration &&
+            equalsIgnoringDuration(that)
+    }
+
+    fun equalsIgnoringDuration(other: CertificateGenerationParameters?): Boolean {
+        return keyLength == other!!.keyLength &&
+            isSelfSigned == other.isSelfSigned &&
+            isCa == other.isCa &&
+            (caName == other.caName || caName == null || other.caName == null) &&
```

Comment:
I don't understand this line. Why isn't it just
```
caName == other.caName
```

like all other lines are tested?
If this logic is correct, I would like to somehow see some code that explains that logic, for example by extracting a method with a good name.

Reply:
FYI this line is in the existing code. This PR only moves this line by extracting a method.

Reply:
Ah, I see. I was wondering about that :).